### PR TITLE
Enable all compile warnings and fix errors

### PR DESCRIPTION
**Pull request by tvegas1**
## What
Enable all warnings and fix issues.

## How
Tested: clang-12, gcc-4.8.5, `--with-sharp` and `--with-ucx`, with/without `--enable-debug` and `ENABLE_TIMER` set.

### CODE REVIEW

**Comment by tvegas1** [comment]
pls look @brminich

File: src/ucx_rma_plugin.c [hunk 1]
```diff
@@ -425,22 +420,28 @@ static ucs_status_t nccl_ucx_rma_am_rkey_cb(void *arg, void *data, size_t length
...
-                              &comm->rkeys[rkey_buf->index].rkey));
-  return UCS_OK;
+  status = ucp_ep_rkey_unpack(comm->ep, rkey_buf->buf,
+                              &comm->rkeys[rkey_buf->index].rkey);
+  if (status != UCS_OK) {
+    WARN("Failed: UCX am rkey cb: rkey unpack error %s",
+         ucs_status_string(status));
+  }
+
+  return status;
```
Comment (brminich) [inline src/ucx_rma_plugin.c:436]: According to UCP AM API, AM callback maybe return either `UCS_OK` or `UCS_INPROGRESS`.
The former means that data is no longer needed and can be released, the latter means that data will need to be accessed later and should not be released.
So, need to return `UCS_OK` in any case here

Comment (tvegas1) [reply src/ucx_rma_plugin.c]: fixed with new behavior

**Review by brminich — APPROVED** [review]

File: src/ucx_plugin.c [hunk 2]
```diff
@@ -401,12 +396,11 @@ static void ucx_request_init(ucx_comm_t *comm) {
 }
 
 ncclResult_t nccl_ucx_connect(int dev, void *handle, void **send_comm, ncclNetDeviceHandle_t** sendDevComm) {
-  ucx_listen_handle_t  *recv_handle = (ucx_listen_handle_t*)handle;
-  struct ncclUCXCommStage* stage = &recv_handle->stage;
-  ucx_comm_t           *comm;
+  ucx_listen_handle_t     *recv_handle = (ucx_listen_handle_t*)handle;
+  struct ncclUCXCommStage *stage       = &recv_handle->stage;
+  ucx_comm_t              *comm        = stage->comm;
```
Comment (tvegas1) [inline src/ucx_plugin.c:401]: this one fixes crash seen at scale

**Comment by brminich** [comment]
the failure seems to be unrelated
```
  mpirun -x LD_LIBRARY_PATH -x NCCL_DEBUG=INFO -x NCCL_DEBUG_SUBSYS=INIT -x UCX_MEMTYPE_CACHE=n -x HCOLL_ENABLE_SHARP=0 -x HCOLL_ENABLE_MCAST_ALL=0 -mca pml ucx -mca coll_hcoll_enable 1 --map-by node --bind-to none --hostfile /GIT/nccl-rdma-sharp-plugins/.ci/cfg/nccl-sharp-1/hostfile -np 2 --report-bindings --allow-run-as-root -mca oob_tcp_if_exclude eth0 -x NCCL_PLUGIN_P2P=ib /GIT/nccl-rdma-sharp-plugins/.ci/nccl_tests /GIT/ml-nccl-rdma-sharp-plugins-pr/1234/nccl-tests/build/broadcast_perf -b 8 -e 4M -f 2 -g 1 -c 1 -z 1 -n 100 -w 100 -p 0 
  --------------------------------------------------------------------------
  ORTE was unable to reliably start one or more daemons.
  This usually is caused by:
```

File: src/ucx_plugin.c (hunk 2 above)
Comment (tvegas1) [reply src/ucx_plugin.c:401]: was merged as #129

**Comment by tvegas1** [comment]
> ORTE was unable to reliably start one or more daemons.

Will check that since it seems consistent on this PR only.

**Review by brminich — APPROVED** [review]

**Review by bureddy — APPROVED** [review]

**Pull request merged**